File: frontend/src/types/browser-session.ts
export type BrowserSessionCreateRequest = {
  timezone?: string;
  locale?: string;
};

export type BrowserSessionCreateResponse = {
  browser_session_id: string;
  user_id: string | null;
  timezone: string | null;
  locale: string | null;
  status: "active";
  created_at: string;
};

File: backend/app/schemas/browser_session.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


BrowserSessionStatus = Literal["active"]


class BrowserSessionCreateRequest(BaseModel):
    timezone: str | None = Field(default=None, min_length=1, max_length=80)
    locale: str | None = Field(default=None, min_length=2, max_length=35)


class BrowserSessionCreateResponse(BaseModel):
    browser_session_id: str
    user_id: str | None
    timezone: str | None
    locale: str | None
    status: BrowserSessionStatus
    created_at: datetime
